File: Programming Basics with JS/05. While Loop - Exercise/examPrep.js
function solve (input){
    let allowedFails = Number(input.shift());
    let fails = 0;
    let sumOfGrades = 0;
    let totalTasks = 0;

    let command = input.shift();
    let currentGrade = Number(input.shift());
    let lastTask = " ";

    while(command !== "Enough"){
        if(currentGrade<=4){
            fails++;
        }

        if(fails>=allowedFails){
            console.log(`You need a break, ${fails} poor grades.`);
            break;
        }

        sumOfGrades += currentGrade;
        totalTasks++;
        lastTask = command;


        command = input.shift();
        currentGrade = Number(input.shift());
    }

    if(fails !== allowedFails){
    console.log(`Average score: ${(sumOfGrades/totalTasks).toFixed(2)}`);
    console.log(`Number of problems: ${totalTasks}`);
    console.log(`Last problem: ${lastTask}`);
    }
}

solve(["3",
"Money",
"6",
"Story",
"4",
"Spring Time",
"5",
"Bus",
"6",
"Enough"])

File: Programming Basics with Java/05 While Loop - Exercise/src/ExamPrep.java
import java.util.Scanner;

public class ExamPrep {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int maxFails = Integer.parseInt(scan.nextLine());
        String lastTask = "";
        double scores = 0;
        int amount = 0;
        int fails = 0;
        boolean hasFailed = true;

        while (maxFails > fails){
            String task = scan.nextLine();
            if(task.equals("Enough")){
                hasFailed = false;
                break;
            }

            int currentGrade = Integer.parseInt(scan.nextLine());
            if(currentGrade <= 4){
                fails++;
            }

            scores += currentGrade;
            amount++;
            lastTask = task;
        }

        if(hasFailed){
            System.out.printf("You need a break, %d poor grades.", fails);
        } else {
            System.out.printf("Average score: %.2f%n", scores / amount);
            System.out.printf("Number of problems: %d%n", amount);
            System.out.printf("Last problem: %s", lastTask);
        }
    }
}